lib/exercism/code_analyzer.rb

```diff
@@ -42,6 +61,54 @@ def run
 
 class Java < CodeAnalyzer
   def run
-    "---java---"
+   projectName = user.username + user.id.to_s + git_rep_info.split('/').last
+   xml = "<project>
+   <actions/><description/>
+   <keepDependencies>false</keepDependencies>
+   <properties><com.coravy.hudson.plugins.github.GithubProjectProperty plugin='github@1.11.3'>
+   <projectUrl>https://github.com/#{git_rep_info}</projectUrl>
+   </com.coravy.hudson.plugins.github.GithubProjectProperty></properties><scm class='hudson.plugins.git.GitSCM' plugin='git@2.3.5'><configVersion>2</configVersion>
+   <userRemoteConfigs><hudson.plugins.git.UserRemoteConfig>
+    <url>https://github.com/#{git_rep_info}.git</url></hudson.plugins.git.UserRemoteConfig>
+    </userRemoteConfigs>
+    <branches><hudson.plugins.git.BranchSpec><name>*/master</name></hudson.plugins.git.BranchSpec></branches>
+    <doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations>
+    <submoduleCfg class='list'/><extensions/></scm>
+    <canRoam>true</canRoam><disabled>false</disabled><blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+    <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding><triggers/>
+    <concurrentBuild>false</concurrentBuild><builders>
+    <hudson.plugins.sonar.SonarRunnerBuilder plugin='sonar@2.2.1'><project/>
+    <properties># required metadata
+      sonar.projectKey=#{projectName}
+      sonar.projectName=#{projectName}
+      sonar.projectVersion=2.0
+      # path to source directories (required)
+      sonar.sources=src
+    </properties>
+      <javaOpts/>
+    <jdk>(Inherit From Job)</jdk><task/>
+    </hudson.plugins.sonar.SonarRunnerBuilder></builders>
+    <publishers/><buildWrappers/></project>"
+
+    jobs = RestClient.get "#{JENKINS_URL}/api/json?tree=jobs[name]", {:content_type => "application/json"}
```

Comment:
Line is too long. [106/80]<br>Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Inconsistent indentation detected.<br>Space inside { missing.<br>Space inside } missing.